ios/FluentUI/Navigation/BarButtonItemView.swift

```diff
@@ -0,0 +1,269 @@
+//
+//  Copyright (c) Microsoft Corporation. All rights reserved.
+//  Licensed under the MIT License.
+//
+
+import UIKit
+
+class BarButtonItemView: UIView {
+    let item: UIBarButtonItem
+    let isLeftItem: Bool
+
+    init(item: UIBarButtonItem, isLeftItem: Bool) {
+        self.item = item
+        self.isLeftItem = isLeftItem
+        super.init(frame: .zero)
+
+        container.addArrangedSubview(button)
+        container.translatesAutoresizingMaskIntoConstraints = false
+        addSubview(container)
+
+        container.addSubview(badgeView)
+
+        isAccessibilityElement = true
+        updateAccessibilityLabel()
+
+        showsLargeContentViewer = true
+        scalesLargeContentImage = true
+
+        accessibilityHint = item.accessibilityHint
+        accessibilityIdentifier = item.accessibilityIdentifier
+
+        NSLayoutConstraint.activate([
+            container.centerXAnchor.constraint(equalTo: centerXAnchor),
+            container.centerYAnchor.constraint(equalTo: centerYAnchor)
+        ])
+
+        NotificationCenter.default.addObserver(self,
+                                               selector: #selector(badgeValueDidChange),
+                                               name: UIBarButtonItem.badgeValueDidChangeNotification,
+                                               object: item)
+
+        badgeValue = item.badgeValue
+        updateLayout()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        preconditionFailure("init(coder:) has not been implemented")
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
+
+        // Need to layout container subviews now, otherwise imageView's frame will get updated later
+        container.layoutSubviews()
+        buttonFrame = button.frame
+    }
+
+    open override var intrinsicContentSize: CGSize {
+        return sizeThatFits(CGSize(width: CGFloat.greatestFiniteMagnitude, height: CGFloat.greatestFiniteMagnitude))
+    }
+
+    open override func sizeThatFits(_ size: CGSize) -> CGSize {
+        let size = container.systemLayoutSizeFitting(UIView.layoutFittingCompressedSize)
+        return size
+    }
+
+    override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
+        super.traitCollectionDidChange(previousTraitCollection)
+        if previousTraitCollection?.horizontalSizeClass != traitCollection.horizontalSizeClass || previousTraitCollection?.verticalSizeClass != traitCollection.verticalSizeClass {
```

Comment:
do not compare with an optional value, first unwrapped it and do compare.

Reply:
What is the problem here? If the optional is nil then this condition will be true and that's what I actually want. Also, there will be unnecessary lines of code for unwrapping and then comparing it.